app/src/main/java/com/example/riden/Adapters/RideAdapter.java

```diff
@@ -83,13 +104,70 @@ public void onClick(View v) {
                 user.saveInBackground();
             }
         });
+
     }
 
     @Override
     public int getItemCount() {
         return rides.size();
     }
 
+    @Override
+    public Filter getFilter() {
+        return filter;
+    }
+
+    Filter filter = new Filter() {
+
+       //TODO: Implement trie data structure search
+
+        /**second complicated feature: type location, but your going in that radius
+         * apart from just the name of the location, show geographical coordintaes
+         * and see which rides going to that destination are close enough
+         * - AD Tree
+         * - geographical search Tree
+         * */
+        @Override
+        protected FilterResults performFiltering(CharSequence constraint) {
+            List<Ride> filteredList = new ArrayList<>();
+            if (constraint.toString().isEmpty()) {
+                filteredList.addAll(allRides);
+            }
+            else {
+                for (Ride ride: allRides) {
+                    if(ride.getDestinationAddress().toLowerCase().contains(constraint.toString().toLowerCase())) {
+                        filteredList.add(ride);
+                        Log.i("filter", ride.getDestinationAddress());
+                    }
+                }
+            }
+            FilterResults filterResults = new FilterResults();
+            filterResults.values = filteredList;
+//            Log.i("filter", filterResults.toString());
+//            notifyDataSetChanged();
+            return filterResults;
+        }
+
+        @Override
+        protected void publishResults(CharSequence constraint, FilterResults results) {
+            rides.clear();
+            Log.i("I", "am being claled");
```

Comment:
nit: Make log statements more descriptive. Look at other examples on Github to see useful log statements